tests/POC/equil_center.py

```diff
@@ -0,0 +1,134 @@
+import random
+import matplotlib.pyplot as plt
+import numpy as np
+import concurrent.futures
+
+DISTANCE_TO_ORIGIN_WEIGHT = 1
+
+class Drone:
+    def __init__(self, position=(0.0,0.0)):
+        self.x, self.y = position
+        self.intended_direction = (0, 0)
+        self.current_cost = float('inf')  # Initialize with a high cost
+
+    def calculate_direction(self, all_drones, drone_index, D):
+        min_value = float('inf')
+        best_dir = (0, 0)
+
+        for dx in [-1, 0, 1]:
+            for dy in [-1, 0, 1]:
+                # Calculate average distance of all drones from the origin excluding the current drone and including drone's potential new position
+                cost = DISTANCE_TO_ORIGIN_WEIGHT*(sum([np.sqrt(drone.x**2 + drone.y**2) for i, drone in enumerate(all_drones) if i != drone_index]) + np.sqrt((self.x + dx)**2 + (self.y + dy)**2)) / len(all_drones)
+
+                for j, other_drone in enumerate(all_drones):
+                    if j != drone_index:
+                        dist = np.sqrt((self.x + dx - other_drone.x)**2 + (self.y + dy - other_drone.y)**2)
+                        cost += (dist - D[drone_index][j])**2
+
+                if cost < min_value:
+                    min_value = cost
+                    best_dir = (dx, dy)
+
+        self.intended_direction = best_dir
+        self.current_cost = min_value
+
+    def move(self):
+        self.x += self.intended_direction[0]
+        self.y += self.intended_direction[1]
+
+def worker(drone, all_drones, idx, D):
+    drone.calculate_direction(all_drones, idx, D)
+    return drone
+
+def individual_drone_score(drone, all_drones, drone_index, D):
+    score = sum([np.sqrt(d.x**2 + d.y**2) for d in all_drones]) / len(all_drones)
+    
+    # Sum of squared differences from the desired distances
+    for j, other_drone in enumerate(all_drones):
+        if j != drone_index:
+            dist = np.sqrt((drone.x - other_drone.x)**2 + (drone.y - other_drone.y)**2)
+            score += (dist - D[drone_index][j])**2
+    
+    return score
+
+
+def individual_drone_score(drone, all_drones, drone_index, D):
+    score = 0
+    
+    # Drone's distance from the origin
+    score += sum([np.sqrt(d.x**2 + d.y**2) for d in all_drones]) / len(all_drones)
+    
+    # Sum of squared differences from the desired distances
+    for j, other_drone in enumerate(all_drones):
+        if j != drone_index:
+            dist = np.sqrt((drone.x - other_drone.x)**2 + (drone.y - other_drone.y)**2)
+            score += (dist - D[drone_index][j])**2
+    
+    return score
+
+
+def simulate_environment(N, D):
+    x_center, y_center = 50, 50
```

Comment:
Avoid single letter variable names -- they're hard to track a bit if reading the code for the first time.